pkg/apis/certificates/validation/validation.go

```diff
@@ -572,3 +580,256 @@ func validateTrustBundle(path *field.Path, in string) field.ErrorList {
+	if opts.SkipCryptoValidation {
+		return allErrors
+	}
+
+	// TODO(KEP-4317): Check length of PKIXPublicKey and ProofOfPossession
+	// fields.
+
+	pubAny, err := x509.ParsePKIXPublicKey(req.Spec.PKIXPublicKey)
+	if err != nil {
+		allErrors = append(allErrors, field.Invalid(pkixPath, req.Spec.PKIXPublicKey, "must be a valid PKIX-serialized public key"))
+		return allErrors
+	}
+
+	// Verify public key properties and the proof-of-possession signature.
+
+	switch pub := pubAny.(type) {
+	case ed25519.PublicKey:
+		// ed25519 has no key configuration to check
+		if !ed25519.Verify(pub, []byte(req.Spec.PodUID), req.Spec.ProofOfPossession) {
+			allErrors = append(allErrors, field.Invalid(popPath, req.Spec.ProofOfPossession, "could not verify proof-of-possession signature"))
+			return allErrors
+		}
+
+	case *ecdsa.PublicKey:
+		if pub.Curve != elliptic.P256() && pub.Curve != elliptic.P384() {
+			allErrors = append(allErrors, field.Invalid(pkixPath, req.Spec.PKIXPublicKey, "elliptic public keys must use curve P256 or P384"))
+			return allErrors
+		}
+		if !ecdsa.VerifyASN1(pub, hashBytes([]byte(req.Spec.PodUID)), req.Spec.ProofOfPossession) {
+			allErrors = append(allErrors, field.Invalid(popPath, req.Spec.ProofOfPossession, "could not verify proof-of-possession signature"))
+			return allErrors
+		}
+
+	case *rsa.PublicKey:
+		if pub.Size() != 3072 && pub.Size() != 4096 {
+			allErrors = append(allErrors, field.Invalid(pkixPath, req.Spec.PKIXPublicKey, "RSA keys must have modulus size 3072 or 4096"))
+			return allErrors
+		}
+		if err := rsa.VerifyPKCS1v15(pub, crypto.SHA256, hashBytes([]byte(req.Spec.PodUID)), req.Spec.ProofOfPossession); err != nil {
+			allErrors = append(allErrors, field.Invalid(popPath, req.Spec.ProofOfPossession, "could not verify proof-of-possession signature"))
+			return allErrors
+		}
+
+	default:
+		allErrors = append(allErrors, field.Invalid(pkixPath, req.Spec.PKIXPublicKey, "unknown public key type; supported types are Ed25519, ECDSA, and RSA"))
+		return allErrors
+	}
+
+	// Bail out if the certificate chain hasn't been set.
+	if req.Status.CertificateChain == "" {
+		return allErrors
+	}
+
+	// TODO(KEP-4317): Check length of CertificateChain field.
+
+	leafBlock, rest := pem.Decode([]byte(req.Status.CertificateChain))
+	if leafBlock == nil {
+		allErrors = append(allErrors, field.Invalid(certChainPath, req.Status.CertificateChain, "issued certificate chain must contain at least one certificate"))
+		return allErrors
+	}
+	if leafBlock.Type != "CERTIFICATE" {
+		allErrors = append(allErrors, field.Invalid(certChainPath, req.Status.CertificateChain, "issued certificate chain must consist entirely of CERTIFICATE PEM blocks"))
+		return allErrors
+	}
+
+	leafCert, err := x509.ParseCertificate(leafBlock.Bytes)
+	if err != nil {
+		allErrors = append(allErrors, field.Invalid(certChainPath, req.Status.CertificateChain, "leaf certificate does not parse as valid X.509"))
+		return allErrors
+	}
```

Comment:
@enj can you be sure this does the validation we desire?

Reply:
Actually, for the entire status bits.  Be sure you're  happy.  Discussion suggests it's not as obvious as I might have thought.